gui/wxpython/gui_core/menu.py

```diff
@@ -305,3 +307,150 @@ def OnItemSelected(self, node):
             label = data['description']
 
         self.showNotification.emit(message=label)
+
+
+class RecentFilesMenu:
+    """Add recent files history menu
+
+    Signal FileRequested is emitted if you request file from recent
+    files menu
+    :param str path: file path you requested
+    :param bool file_exists: file path exists or not
+    :param obj file_history: file history obj instance
+
+
+    :param str app_name: required for group name of recent files path
+    written into the .recent_files file
+    :param obj parent_menu: menu widget instance where be inserted
+    recent files menu on the specified position
+    :param int pos: position (index) where insert recent files menu in
+    the parent menu
+    :param int history_len: the maximum number of file paths written
+    into the .recent_files file to app name group
+    """
+
+    recent_files = '.recent_files'
+
+    def __init__(self, app_name, parent_menu, pos, history_len=10):
+        self._history_len = history_len
+        self._parent_menu = parent_menu
+        self._pos = pos
+
+        self.file_requested = Signal('RecentFilesMenu.FileRequested')
+
+        self.filehistory = wx.FileHistory(maxFiles=history_len)
+        # Recent files path stored in GRASS GIS config dir in the
+        # .recent_files file in the group by application name
+        self.config = wx.FileConfig(
+            style=wx.CONFIG_USE_LOCAL_FILE,
+            localFilename=os.path.join(
+                utils.GetSettingsPath(), self.recent_files,
+            ),
+        )
+        self.config.SetPath(strPath=app_name)
+        self.filehistory.Load(self.config)
+        self.RemoveFileFromHistory(
+            stop=self.filehistory.GetCount(),
+            check_if_files_exists=True,
+        )
```

Comment:
Here, expressing what you mean by a function name would be immediately clear to the reader:

```
self.RemoveNonExistentFiles()
```